### PR TITLE
r/sagemaker_space - support app lifecycle management

### PR DESCRIPTION
**Pull request by DrFaust92**
### Relations

Closes #0000

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccSageMaker_serial/Space/jupyterLabAppSettingsLifecycle  PKG=sagemaker

--- PASS: TestAccSageMaker_serial/Space/basic (248.37s)
--- PASS: TestAccSageMaker_serial/Space/jupyterLabAppSettingsLifecycle (278.43s)
--- PASS: TestAccSageMaker_serial/Space/jupyterLabAppSettings (280.42s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
@DrFaust92 Could you add a CHANGELOG entry please?

**Comment by DrFaust92** [comment]
@ewbankkit seems i did it already but forgot to push 😊

File: website/docs/r/sagemaker_space.html.markdown [hunk 1]
```diff
@@ -61,6 +61,8 @@ The `space_sharing_settings` block supports the following argument:
 
 The `code_editor_app_settings` block supports the following argument:
 
+* `app_lifecycle_management` - (Optional) Settings that are used to configure and manage the lifecycle of JupyterLab applications in a space. See [`app_lifecycle_management` Block](#app_lifecycle_management-block) below.
+* `code_repository` - (Optional) A list of Git repositories that SageMaker automatically displays to users for cloning in the JupyterServer application. See [`code_repository` Block](#code_repository-block) below.
```
Comment (DrFaust92) [inline website/docs/r/sagemaker_space.html.markdown:65]: this was missing

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccSageMaker_serial/^Space$$' PKG=sagemaker
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.2 test ./internal/service/sagemaker/... -v -count 1 -parallel 20  -run=TestAccSageMaker_serial/^Space$ -timeout 360m
2024/10/21 08:24:31 Initializing Terraform AWS Provider...
=== RUN   TestAccSageMaker_serial
=== PAUSE TestAccSageMaker_serial
=== CONT  TestAccSageMaker_serial
=== RUN   TestAccSageMaker_serial/Space
=== RUN   TestAccSageMaker_serial/Space/disappears
=== RUN   TestAccSageMaker_serial/Space/jupyterServerAppSettings
=== RUN   TestAccSageMaker_serial/Space/jupyterLabAppSettings
=== RUN   TestAccSageMaker_serial/Space/jupyterLabAppSettingsLifecycle
=== RUN   TestAccSageMaker_serial/Space/basic
=== RUN   TestAccSageMaker_serial/Space/tags
=== RUN   TestAccSageMaker_serial/Space/kernelGatewayAppSettings
=== RUN   TestAccSageMaker_serial/Space/kernelGatewayAppSettings_lifecycleConfig
=== RUN   TestAccSageMaker_serial/Space/kernelGatewayAppSettings_imageConfig
    space_test.go:203: Environment variable SAGEMAKER_IMAGE_VERSION_BASE_IMAGE is not set
=== RUN   TestAccSageMaker_serial/Space/codeEditorAppSettings
=== RUN   TestAccSageMaker_serial/Space/storageSettings
=== RUN   TestAccSageMaker_serial/Space/customFileSystem
--- PASS: TestAccSageMaker_serial (2855.65s)
    --- PASS: TestAccSageMaker_serial/Space (2855.65s)
        --- PASS: TestAccSageMaker_serial/Space/disappears (276.56s)
        --- PASS: TestAccSageMaker_serial/Space/jupyterServerAppSettings (255.28s)
        --- PASS: TestAccSageMaker_serial/Space/jupyterLabAppSettings (272.39s)
        --- PASS: TestAccSageMaker_serial/Space/jupyterLabAppSettingsLifecycle (275.14s)
        --- PASS: TestAccSageMaker_serial/Space/basic (255.02s)
        --- PASS: TestAccSageMaker_serial/Space/tags (252.81s)
        --- PASS: TestAccSageMaker_serial/Space/kernelGatewayAppSettings (214.71s)
        --- PASS: TestAccSageMaker_serial/Space/kernelGatewayAppSettings_lifecycleConfig (215.95s)
        --- SKIP: TestAccSageMaker_serial/Space/kernelGatewayAppSettings_imageConfig (0.00s)
        --- PASS: TestAccSageMaker_serial/Space/codeEditorAppSettings (216.87s)
        --- PASS: TestAccSageMaker_serial/Space/storageSettings (255.37s)
        --- PASS: TestAccSageMaker_serial/Space/customFileSystem (365.57s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker	2860.767s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

**Pull request merged**